src/diepvries/table.py

```diff
@@ -41,31 +96,15 @@ def __init__(self, schema: str, name: str, fields: List[Field], *_args, **_kwarg
             _args: Unused here, useful for children classes.
             _kwargs: Unused here, useful for children classes.
         """
-        self.schema = schema
-        self.name = name.lower()
+        super().__init__(schema=schema, name=name)
         self.fields = fields
 
         # Check if table structure is valid. Each subclass has its own implementation
         # (with its specific tests + the tests performed in this abstract class).
         self._validate()
 
-        # Variables set in DataVaultLoad
-        self.staging_schema = None
-        self.staging_table = None
-
-        self._logger = FixedPrefixLoggerAdapter(logging.getLogger(__name__), str(self))
-
-        self._logger.info("Instance of (%s) created", type(self))
-
-    def __str__(self) -> str:
-        """Representation of a Table object as a string.
-
-        This helps the tracking of logging events per entity.
-
-        Returns:
-            String representation of a Table.
-        """
-        return f"{type(self).__name__}: {self.schema}.{self.name}"
+        # Set in DataVaultLoad
+        self.staging_table: StagingTable = None
```

Comment:
What about declaring `staging_table: StagingTable` at the class level, to avoid the potential type issues with `None`?

Reply:
Good idea, just applied it.